### PR TITLE
Collection of refactors and minor changes

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
I would also recommend to switch away from `augeasproviders_sysctl`/`augeasproviders_core` to `puppet-sysctl` instead. Didn't looked into the details yet, but I am sure it will be possible.

File: manifests/files.pp [hunk 1]
```diff
@@ -47,78 +47,75 @@
...
-            margin                 => $resize_margin,
-            expected_swapfile_size => $swapfilesize,
-            actual_swapfile_size   => $facts['swapfile_sizes'][$swapfile],
-            verbose                => $resize_verbose,
-            before                 => Exec["Create swap file ${swapfile}"],
-          }
-        }
+    # Check for resizing the swap file
+    if $resize_existing and $facts['swapfile_sizes'] {
+      # use $swapfile_sizes for new or $swapfile_sizes_csv as fallback for old Puppet clients
```
Comment (anders-larsson) [inline manifests/files.pp:64]: It feels like something is either missing here is is incorrect. First we're checking if $facts['swapfile_exists'] exists and then on line 65 we grab information from $facts['swapfile_sizes_csv']?

Am I missing something here? Shouldn't we check if swapfile_sizes_csv exists as well or is that part handled by the function that is called?

File: manifests/init.pp [hunk 2]
```diff
@@ -46,22 +46,12 @@
   Hash    $files             = {},
   Boolean $files_hiera_merge = false,
 ) {
-  # variable handling
-  if $files_hiera_merge =~ Boolean {
-    $files_hiera_merge_bool = $files_hiera_merge
-  } else {
-    $files_hiera_merge_bool = str2bool($files_hiera_merge)
+  case $files_hiera_merge {
```
Comment (anders-larsson) [inline manifests/init.pp:49]: Maybe for a later pull request, but should we remove module Hiera merge support altogether as we have done previously and rely on native Hiera merge support?

Might require a new major release though(?)

**Comment by anders-larsson** [comment]
> I would also recommend to switch away from `augeasproviders_sysctl`/`augeasproviders_core` to `puppet-sysctl` instead. Didn't looked into the details yet, but I am sure it will be possible.

That might indeed be sane.

File: manifests/files.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/files.pp:64]: I can only guess here:
both facts `swapfile_sizes` and `swapfile_sizes_csv` are based on the existence of `/proc/swaps`. Either the author is sure both or non will exist or he didn't care for old Puppet clients that much (much like I do :)

We could remove the support for old Puppet clients completely.

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/init.pp:49]: Aggreed, did want to have at least one version that is backward compatible :)

**Comment by Phil-Friderici** [comment]
Let's merge this and create a new PR with the backward breaking stuff we mentioned above ?

**Comment by Phil-Friderici** [comment]
Released as v4.1.0-forked

**Comment by Phil-Friderici** [comment]
Find the other changes in https://github.com/kodguru/petems-swap_file/pull/4

**Pull request merged**